src/renderer/contexts/GlobalNodeState.tsx

```diff
@@ -582,10 +582,14 @@ export const GlobalProvider = memo(
 
         const releaseNodeFromParent = useCallback(
             (id: string) => {
+                let parentNode: Node<NodeData> | undefined;
+                let node: Node<NodeData> | undefined;
+                let capturedNodes: Node<NodeData>[];
```

Comment:
You assumed that state updates will be processed in order. While this guaranteed for a single memory cell, I don't think it's guaranteed for different memory cells. `changeNodes` and `changeEdges` ultimately use `useNodesState`'s and `useEdgesState`'s setters, which probably use different cells.

Reply:
so what do you recommend I do here? use getNodes in the changeEdges callback?

Reply:
I'm pretty sure using `getNodes` inside a state update isn't defined either. I think we need to `getNodes()` and figure out what edges to remove before `changeEdges`.